app/assets/javascripts/common/reports.js

```diff
@@ -1253,9 +1394,70 @@ function withBaseBarConfig(config) {
   );
 }
 
+function mergeConfig(baseConfig, overwritingConfig) {
+  return _.mergeWith(
+    baseConfig,
+    overwritingConfig,
+    mergeArraysWithConcatenation
+  );
+}
 
 function mergeArraysWithConcatenation(objValue, srcValue) {
   if (_.isArray(objValue)) {
     return objValue.concat(srcValue);
   }
 }
+
+function isGoalLineEnabled(domElement) {
+  const cardNodeBPControlled = domElement;
+  const isEnabled = cardNodeBPControlled.querySelector("[data-show-goal-lines]")
+    .dataset.showGoalLines;
+  if (isEnabled === "true") {
+    return true;
+  }
+  return false;
+}
+
+function addGoalLineToConfig(config, periodValues, goalDownwards = false) {
+  const latestDecValue = getLatestDecemberValue(periodValues);
+  const goal = calculateGoal(latestDecValue, goalDownwards);
+  const goalLineConfig = {
+    plugins: [goalLinePlugin(goal)],
+  };
+  return mergeConfig(config, goalLineConfig);
+}
+
+function calculateGoal(value, goalDownwards) {
+  if (goalDownwards) {
+    return calculateGoalDownwards(value);
+  }
+  return calculateGoalUpwards(value);
+}
+
+function calculateGoalUpwards(decemeberValue) {
+  const goal =
+    decemeberValue + (100 - decemeberValue) * relativeImprovementRatio();
+  return Math.ceil(goal);
+}
+
+function calculateGoalDownwards(decemeberValue) {
+  const goal = decemeberValue - decemeberValue * relativeImprovementRatio();
+  return Math.floor(goal);
+}
+
+function relativeImprovementRatio() {
+  const defaultRelativeImprovementPercentage = 10;
+  return defaultRelativeImprovementPercentage / 100;
+}
+
+function getLatestDecemberValue(periodValues) {
+  const dateKeysArray = Object.keys(periodValues);
+  const filterDecemberKeys = dateKeysArray.filter((item) => {
+    if (item.includes("Dec")) {
```

Comment:
The `if` is redundant since `item.includes` already returns `true` or `false`. Could do
`const filterDecemberKeys = dateKeysArray.filter((key) => key.includes("Dec"))`

Reply:
good spot 👀 

fixed